From 2477dbea92d4111fb70beaf975bce0eeee9ac8e5 Mon Sep 17 00:00:00 2001
From: Richard Leitner <richard.leitner@skidata.com>
Date: Fri, 1 Jun 2018 14:28:05 +0200
Subject: [PATCH 6/7] several compile fixes

This patch was taken from Debian's libmxml 2.11-1 source:
	07_compile_fixes.patch
---
 mxml.h    |  2 +-
 mxmldoc.c | 10 +++++++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/mxml.h b/mxml.h
index 5ad96b8..e78b8e9 100644
--- a/mxml.h
+++ b/mxml.h
@@ -28,7 +28,7 @@
 #  include <string.h>
 #  include <ctype.h>
 #  include <errno.h>
-
+#  include <stdarg.h>
 
 /*
  * Constants...
diff --git a/mxmldoc.c b/mxmldoc.c
index de5c8a7..bbdf758 100644
--- a/mxmldoc.c
+++ b/mxmldoc.c
@@ -37,7 +37,15 @@ extern char **environ;
 #  include "zipc.h"
 #endif /* HAVE_ZLIB_H */
 
-
+#  ifndef HAVE_STRLCPY
+extern size_t	_mxml_strlcpy(char *, const char *, size_t);
+#    define strlcpy _mxml_strlcpy
+#  endif /* !HAVE_STRLCPY */
+
+#ifndef HAVE_STRLCAT
+extern size_t  _mxml_strlcat( char *, const char *, size_t);
+#define strlcat _mxml_strlcat
+#endif
 /*
  * This program scans source and header files and produces public API
  * documentation for code that conforms to the CUPS Configuration
-- 
2.11.0

